include/lsst/afw/image/TransmissionCurve.h

```diff
@@ -0,0 +1,300 @@
+        double getSpacing() const { return size == 0 ? 0 : (max - min)/size; }
+
+        /// Create an array containing the full set of wavelength sample points.
+        ndarray::Array<double,1,1> makeArray() const;
+    };
+
+    /**
+     *  Create a new TranmissionCurve that has unit thoughput at all wavelengths everywhere.
+     */
+    static std::shared_ptr<TransmissionCurve> makeIdentity();
+
+    /**
+     *  Create a new TransmissionCurve with spatially-constant throughput.
+     *
+     *  @param[in]  throughput      an Array of throughput values with the same
+     *                              size as the wavelengths Array (will be copied).
+     *  @param[in]  wavelengths     an Array of wavelengths in Angstroms (will be copied).
+     *                              Must be monotonically increasing.
+     *  @param[in]  throughputAtMin the throughput value used for wavelengths
+     *                              below wavelengths.front().
+     *  @param[in]  throughputAtMax the throughput value used for wavelengths
+     *                              above wavelengths.back().
+     */
+    static std::shared_ptr<TransmissionCurve> makeConstant(
+        ndarray::Array<double const,1> const & throughput,
+        ndarray::Array<double const,1> const & wavelengths,
+        double throughputAtMin=0.0, double throughputAtMax=0.0
+    );
+
+    /**
+     *  Create a new TransmissionCurve with throughput varying as function of radius.
+     *
+     *  @param[in]  throughput      an Array of throughput values with shape
+     *                              (wavelengths.size(), radii.size()).  Will be
+     *                              copied.
+     *  @param[in]  wavelengths     an Array of wavelengths in Angstroms (will be copied).
+     *                              Must be monotonically increasing.
+     *  @param[in]  radii           an Array of radii (will be copied).
+     *                              Must be monotonically increasing.
+     *  @param[in]  throughputAtMin the throughput value used for wavelengths
+     *                              below wavelengths.front().
+     *  @param[in]  throughputAtMax the throughput value used for wavelengths
+     *                              above wavelengths.back().
+     *
+     *  Throughput at radii below radii.front() or above radii.back() is assumed
+     *  to be constant.
+     */
+    static std::shared_ptr<TransmissionCurve> makeRadial(
+        ndarray::Array<double const,2> const & throughput,
+        ndarray::Array<double const,1> const & wavelengths,
+        ndarray::Array<double const,1> const & radii,
+        double throughputAtMin=0.0, double throughputAtMax=0.0
+    );
+
+    /**
+     *  Return a new TransmissionCurve that simply multiplies the values of two others.
+     *
+     *  The product is typically computed lazily (the returned object samples
+     *  from each operand, and then multiplies those samples).
```

Comment:
Will move the comment.

Because the wavelengths to be evaluated are always provided by the user, the reported grid is really just supposed to indicate the sampling rate and bounds, so the sampling rate is the minimum of the input sampling rates and the bounds are set to the minimum or maximum of the inputs depending on whether one of them goes to zero at a bound.

Reply:
Please document that, or at least a general statement that the new grid may be an intelligent merge of the old ones.